notebook/notebookapp.py

```diff
@@ -2289,16 +2319,19 @@ def list_running_servers(runtime_dir=None):
     if not os.path.isdir(runtime_dir):
         return
 
+    api_version_json_bytes = json.dumps({"version":__version__}).encode()
     for file_name in os.listdir(runtime_dir):
         if re.match('nbserver-(.+).json', file_name):
             with io.open(os.path.join(runtime_dir, file_name), encoding='utf-8') as f:
                 info = json.load(f)
 
-            # Simple check whether that process is really still running
+            # Actively check whether that process is really available via an HTTP request
             # Also remove leftover files from IPython 2.x without a pid field
-            if ('pid' in info) and check_pid(info['pid']):
+            response = kernel_request(info, path=url_path_join(info.get('base_url') or '/','/api'))
+            try:
+                assert JUPYTER_NOTEBOOK_TAG == json.loads(response.body)["module"]
                 yield info
-            else:
+            except:
```

Comment:
This is not sufficient for then the request returns successfully, but doesn't match.

Reply:
@kevin-bates , Approved. really good job!

commit 57330ff68ddaac913c615263fb1240acfd74d1c6 (HEAD -> master, origin/master, origin/HEAD)
Author: git <abaelhe@icloud.com>
Date:   Fri Sep 4 00:37:45 2020 +0800

    let user make choices to their data, preference or environment.

commit 908a630761aa849e0824b2710d61560d7df278d3
Author: git <abaelhe@icloud.com>
Date:   Fri Sep 4 00:20:37 2020 +0800

User Notifications for policy change of list_running_servers()

commit a305e0d57f5cced26201a27ec69a892e8107487a
Author: git <abaelhe@icloud.com>
Date:   Fri Sep 4 00:18:59 2020 +0800

    User Notifications for policy change of list_running_servers()

commit 958151a7b8df4f1a0cd13d96c751a961f416ce38
Author: git <abaelhe@icloud.com>
Date:   Thu Sep 3 22:43:55 2020 +0800